oak/server/oak_manager.cc

```diff
@@ -21,17 +21,32 @@
 #include "gflags/gflags.h"
 
 DEFINE_string(enclave_path, "", "Path to enclave to load");
+DEFINE_uint32(node_port_min, 7000,
+              "Minimum value of port number assigned to a node");
+DEFINE_uint32(node_port_max, 8000,
+              "Maximum value of port number assigned to a node");
 
-OakManager::OakManager() : Service(), node_id_(0) { InitializeEnclaveManager(); }
+OakManager::OakManager()
+    : Service(), node_id_(0), node_port_(FLAGS_node_port_min) {
+  InitializeEnclaveManager();
+}
 
 ::grpc::Status OakManager::CreateNode(::grpc::ServerContext* context,
                                       const ::oak::CreateNodeRequest* request,
                                       ::oak::CreateNodeResponse* response) {
   std::string node_id = NewNodeId();
-  CreateEnclave(node_id, request->module());
+
+  if (node_port_ > FLAGS_node_port_max) {
+    return ::grpc::Status(::grpc::StatusCode::RESOURCE_EXHAUSTED,
+                          "No free port available for new node.");
+  }
+
+  CreateEnclave(node_id, request->module(), node_port_);
   ::oak::InitializeOutput out = GetEnclaveOutput(node_id);
-  response->set_port(out.port());
+  response->set_port(node_port_);
   response->set_node_id(node_id);
+  // Increments the port number.
+  node_port_ += 1;
```

Comment:
This does not seem thread safe?

Reply:
Probably should have used std::atomic here. I thought GRPC dispatch would be single-threaded a. la. Chromium but looks like it's using a thread pool. But the code is obsolete now